### PR TITLE
Delete brach associated with pr

### PR DESCRIPTION
**Pull request by mklauber**
Issue #580 Added a `delete_branch` flag to the `pr.merge(...)` method.  If the flag is true, it checks if the branch is part of other PRs.  If it is not, then it deletes the branch.

### CODE REVIEW

File: github/PullRequest.py [hunk 1]
```diff
@@ -775,6 +775,19 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
             self.url + "/merge",
             input=post_parameters
         )
+
+
+        if delete_branch == True:
```
Comment (sfdye) [inline github/PullRequest.py]:
```suggestion
        if delete_branch:
```

File: github/PullRequest.py [hunk 2]
```diff
@@ -775,6 +775,19 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
...
+
+            # Check for other pull requests referencing the same ref.
+            # If found, abort.
+            remaining_pulls = self.head.repo.get_pulls(head=self.head.ref)
+            if remaining_pulls > 0:
```
Comment (sfdye) [inline github/PullRequest.py]:
```suggestion
            if remaining_pulls:
```
I guess this will work already

**Comment by sfdye** [comment]
Looks good! I will be more convinced if a unit test can be added. See:
https://github.com/PyGithub/PyGithub/blob/master/CONTRIBUTING.md#automated-tests

File: github/PullRequest.py [hunk 3]
```diff
@@ -775,6 +775,19 @@ def merge(self, commit_message=github.GithubObject.NotSet, commit_title=github.G
...
             input=post_parameters
         )
+
+
+        if delete_branch:
+
+            # Check for other pull requests referencing the same ref.
+            # If found, abort.
+            remaining_pulls = self.head.repo.get_pulls(head=self.head.ref)
+            if remaining_pulls:
```
Comment (sfdye) [inline github/PullRequest.py]: What if I want to force delete a branch? Then `delete_branch=True` will not work, if some other PR is still referencing it right? If this the intended behavior, I am not so sure.

Comment (mklauber) [reply github/PullRequest.py]: Not sure.  I don't think merging one PR should close other PRs without explicit confirmation. 
 Perhaps deleting a branch should be done via a separate call?  So I take the code I added and put it in a `delete_branch(force=False)` method? 

So typical usage would be
```
pr = repo.get_pull(...)
pr.merge(...)
pr.delete_branch(force=True)
```

Comment (mklauber) [reply github/PullRequest.py]: Actually that would make a lot of sense, because then the api for restoring a branch is similar.

**Comment by jwillenbring** [comment]
This would be a very useful feature.

**Pull request closed without merging**